frontend/src/ducks/transactions.ts

```diff
@@ -0,0 +1,91 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+
+import {
+  ledgerTransactionHistoryConfig,
+  networkConfig,
+} from "constants/settings";
+import {
+  TransactionsInitialState,
+  Network,
+  RejectMessage,
+  ActionStatus,
+  FetchTransactionsHistoryActionResponse,
+} from "types";
+import { RootState } from "config/store";
+import { getErrorString } from "helpers/getErrorString";
+import { parseDateFromFormat } from "helpers/parseDateFromFormat";
+
+export const fetchTransactionsHistoryAction = createAsyncThunk<
+  FetchTransactionsHistoryActionResponse,
+  Network,
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "transactions/fetchTransactionsHistoryAction",
+  async (network, { rejectWithValue }) => {
+    try {
+      const historyFilter = ledgerTransactionHistoryConfig["30D"];
+
+      const ledgersUrlRequest = `/api/ledgers${historyFilter.endpointPrefix}${networkConfig[network].ledgerTransactionsHistorySuffix}`;
+      const response = await fetch(ledgersUrlRequest);
+
+      const { data: ledgers } = await response.json();
+
+      return {
+        items: ledgers.map((ledger: Record<string, unknown>) => ({
```

Comment:
let's include the `end` data here too (see my other comment for more details why)